src/OperatorEnumConstruction.jl

```diff
@@ -51,108 +52,159 @@ function create_evaluation_helpers!(operators::GenericOperatorEnum)
+            end
+            function $($f)(l::T, r::Node{T}) where {T<:$($type_requirements)}
+                if r.degree == 0 && r.constant
+                    Node(T; val=$($f)(l, r.val::T))
+                else
+                    Node($($op), Node(T; val=l), r)
                 end
-
+            end
+            if $($build_converters)
                 # Converters:
-                function $f(
+                function $($f)(
                     l::Node{T1}, r::Node{T2}
-                ) where {T1<:$type_requirements,T2<:$type_requirements}
+                ) where {T1<:$($type_requirements),T2<:$($type_requirements)}
                     T = promote_type(T1, T2)
                     l = convert(Node{T}, l)
                     r = convert(Node{T}, r)
-                    return $f(l, r)
+                    return $($f)(l, r)
                 end
-                function $f(
+                function $($f)(
                     l::Node{T1}, r::T2
-                ) where {T1<:$type_requirements,T2<:$type_requirements}
+                ) where {T1<:$($type_requirements),T2<:$($type_requirements)}
                     T = promote_type(T1, T2)
                     l = convert(Node{T}, l)
                     r = convert(T, r)
-                    return $f(l, r)
+                    return $($f)(l, r)
                 end
-                function $f(
+                function $($f)(
                     l::T1, r::Node{T2}
-                ) where {T1<:$type_requirements,T2<:$type_requirements}
+                ) where {T1<:$($type_requirements),T2<:$($type_requirements)}
                     T = promote_type(T1, T2)
                     l = convert(T, l)
                     r = convert(Node{T}, r)
-                    return $f(l, r)
+                    return $($f)(l, r)
                 end
-            end,
-        )
+            end
+        end
     end
-    # Redefine Base operations:
-    for (op, f) in enumerate(map(Symbol, operators.unaops))
-        if isdefined(Base, f)
-            f = :(Base.$(f))
-        elseif !extend_user_operators
-            # Skip non-Base operators!
-            continue
+end
+
+function _extend_operators(operators, skip_user_operators, __module__::Module)
+    binary_ex = _extend_binary_operator(:f, :op, :type_requirements, :build_converters)
+    unary_ex = _extend_unary_operator(:f, :op, :type_requirements)
+    return quote
+        local type_requirements
+        local build_converters
+        if isa($operators, OperatorEnum)
+            type_requirements = Real
+            build_converters = true
+        else
+            type_requirements = Any
+            build_converters = false
+        end
+        for (op, f) in enumerate(map(Symbol, $(operators).binops))
+            if isdefined(Base, f)
+                f = :(Base.$(f))
+            elseif $(skip_user_operators)
+                continue
+            else
+                f = :($($__module__).$(f))
```

Comment:
This is how user-defined operators are extended - the module name is loaded from the macro.

This seems to work when the custom operator is defined inside a module as well.